Tests fail with K8s 1.10.4

Hit in PR #890 
More diags at https://semaphoreci.com/calico/libcalico-go/branches/pull-request-890/builds/3
```
Failure [1.032 seconds]

Test Syncer API for Kubernetes backend [It] should patch Pod (with existing annotations) with our PodIP annotation 

/go/src/github.com/projectcalico/libcalico-go/lib/backend/k8s/k8s_test.go:1409



  Expected error:

      <*errors.StatusError | 0xc42106e2d0>: {

          ErrStatus: {

              TypeMeta: {Kind: "", APIVersion: ""},

              ListMeta: {SelfLink: "", ResourceVersion: "", Continue: ""},

              Status: "Failure",

              Message: "object is being deleted: pods \"test-syncer-basic-pod\" already exists",

              Reason: "AlreadyExists",

              Details: {

                  Name: "test-syncer-basic-pod",

                  Group: "",

                  Kind: "pods",

                  UID: "",

                  Causes: nil,

                  RetryAfterSeconds: 0,

              },

              Code: 409,

          },

      }

      object is being deleted: pods "test-syncer-basic-pod" already exists

  not to have occurred



  /go/src/github.com/projectcalico/libcalico-go/lib/backend/k8s/k8s_test.go:1317

------------------------------

••••••



Summarizing 3 Failures:



[Fail] Test Syncer API for Kubernetes backend [It] should handle a basic Pod 

/go/src/github.com/projectcalico/libcalico-go/lib/backend/k8s/k8s_test.go:234



[Fail] Test Syncer API for Kubernetes backend [It] should patch Pod (with no existing annotations) with our PodIP annotation 

/go/src/github.com/projectcalico/libcalico-go/lib/backend/k8s/k8s_test.go:1317



[Fail] Test Syncer API for Kubernetes backend [It] should patch Pod (with existing annotations) with our PodIP annotation 

/go/src/github.com/projectcalico/libcalico-go/lib/backend/k8s/k8s_test.go:1317



Ran 16 of 16 Specs in 105.474 seconds

FAIL! -- 13 Passed | 3 Failed | 0 Pending | 0 Skipped --- FAIL: TestModel (105.48s)
```

--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ test: ut
 
 # Define some constants
 #######################
-K8S_VERSION       = v1.8.1
+K8S_VERSION      ?= v1.10.4
 GO_BUILD_VER     ?= v0.12
 CALICO_BUILD     ?= calico/go-build:$(GO_BUILD_VER)
 PACKAGE_NAME     ?= projectcalico/libcalico-go
@@ -187,7 +187,7 @@ run-kubernetes-master: stop-kubernetes-master
 		-v  $(CURDIR):/manifests \
 		lachlanevenson/k8s-kubectl:${K8S_VERSION} \
 		--server=http://127.0.0.1:8080 \
-		apply -f manifests/test/crds.yaml
+		apply -f /manifests/test/crds.yaml
 
 	# Create a Node in the API for the tests to use.
 	docker run \
@@ -196,7 +196,7 @@ run-kubernetes-master: stop-kubernetes-master
 		-v  $(CURDIR):/manifests \
 		lachlanevenson/k8s-kubectl:${K8S_VERSION} \
 		--server=http://127.0.0.1:8080 \
-		apply -f manifests/test/mock-node.yaml
+		apply -f /manifests/test/mock-node.yaml
 
 	# Create Namespaces required by namespaced Calico `NetworkPolicy`
 	# tests from the manifests namespaces.yaml.
@@ -206,7 +206,7 @@ run-kubernetes-master: stop-kubernetes-master
 		-v  $(CURDIR):/manifests \
 		lachlanevenson/k8s-kubectl:${K8S_VERSION} \
 		--server=http://localhost:8080 \
-		apply -f manifests/test/namespaces.yaml
+		apply -f /manifests/test/namespaces.yaml
 
 ## Stop the local kubernetes master
 stop-kubernetes-master:

--- a/lib/backend/k8s/k8s_test.go
+++ b/lib/backend/k8s/k8s_test.go
@@ -1284,7 +1284,12 @@ var _ = Describe("Test Syncer API for Kubernetes backend", func() {
 		})
 
 		By("Deleting the Pod and expecting the wep to be deleted", func() {
-			err = c.ClientSet.CoreV1().Pods("default").Delete(pod.ObjectMeta.Name, &metav1.DeleteOptions{})
+			var zero int64
+			policy := metav1.DeletePropagationBackground
+			err = c.ClientSet.CoreV1().Pods("default").Delete(pod.ObjectMeta.Name, &metav1.DeleteOptions{
+				GracePeriodSeconds: &zero,
+				PropagationPolicy:  &policy,
+			})
 			Expect(err).NotTo(HaveOccurred())
 			cb.ExpectDeleted([]model.KVPair{expectedKVP})
 		})
@@ -1396,7 +1401,12 @@ var _ = Describe("Test Syncer API for Kubernetes backend", func() {
 		})
 
 		By("Deleting the Pod and expecting the wep to be deleted", func() {
-			err = c.ClientSet.CoreV1().Pods("default").Delete(pod.ObjectMeta.Name, &metav1.DeleteOptions{})
+			var zero int64
+			policy := metav1.DeletePropagationBackground
+			err = c.ClientSet.CoreV1().Pods("default").Delete(pod.ObjectMeta.Name, &metav1.DeleteOptions{
+				GracePeriodSeconds: &zero,
+				PropagationPolicy:  &policy,
+			})
 			Expect(err).NotTo(HaveOccurred())
 			cb.ExpectDeleted([]model.KVPair{expectedKVP})
 		})

